app/src/androidTest/java/ch/epfl/sdp/appart/UserAdsActivityTest.java

```diff
@@ -124,7 +127,69 @@ public void userThatCreatedAnAdCanFindHisAds() {
 
         onView(allOf(withIndex(withId(R.id.image_CardLayout_imageView), 0),isDisplayed()));
     }
-    
+
+    @Test
+    public void deletingAdsWork() {
```

Comment:
Please remember to use a `@Before` in the test classes to clear the shared preferences and recreate the activity so that we don't get failing tests because of auto-login. It would be good to also have an `@After` where you clear the sharedpreferences again.

It can be done like this:
```
mScenarioRule.getScenario().onActivity(ac -> {
    SharedPreferencesHelper.clearSavedUserForAutoLogin(ac);
    ac.recreate();
});
```

Reply:
I had already an @After in this class, but no @Before doing what you said indeed. I'll work on that !